chore: update config and docs for creating the `server certificate`

enzo-bitfly opened the pull request:
- re-enable `certificate` usage in local development with `https`
- restructure `setup` section in `readme`

See: BEDS-1109

marcel-bitfly commented on frontend/.gitignore:
```diff
@@ -30,8 +30,7 @@ pnpm-lock.yaml
 
 .npmrc
 
-server.crt
```
**thought:** probably best to keep that in

- if somebody still has `.crt` files in their repository we probably don't want them to commit the files
- you could add a comment why it is still in there if you think it is a good idea

enzo-bitfly replied:
I added it back with a comment

marcel-bitfly commented on frontend/README.md:
````diff
@@ -61,6 +40,30 @@ yarn install
 ```
 
+### Create `.env` file
+Create a `.env` file from the `.env-example` file existing in the repository:
````
**nitpick:** you basically repeat the `headline`

```suggestion
```

enzo-bitfly replied:
I reworded it to avoid repetition but still mention that it's based on `.env-example`

marcel-bitfly commented on frontend/README.md:
````diff
@@ -61,6 +40,30 @@ yarn install
+```
+
+Inside of `.env`, write the URLs of the API servers and the secret key to access to them.
+The variable evoking the development is used to show/hide features and components that are not ready for production.
````
**suggestion:** unclear description
```suggestion
Inside of `.env`, add necessary URLs and secrets.
```

enzo-bitfly replied:
Good one, done

marcel-bitfly commented on frontend/README.md:
````diff
@@ -61,6 +40,30 @@ yarn install
+# create a certificate for your local development domain
+mkcert local.beaconcha.in localhost 127.0.0.1 ::1
+```
+
````
**suggestion:** 
 
### Create `certificates` to enable local development via `https`

We recommend using `mkcert` for creating self-signed certificates, as it simplifies the process and avoids browser warnings. If you don't have `mkcert` installed, you can follow the [installation instructions](https://github.com/FiloSottile/mkcert#installation).

```bash
# e.g. on mac
brew install mkcert
```

Once installed, run:

```bash
# create a local CA (Certificate Authority)
mkcert -install
# create a certificate for your local development domain
mkcert local.beaconcha.in localhost 127.0.0.1 ::1
```

enzo-bitfly replied:
I partially include this suggestion. I leave the example on mac out because this doesn't cover Firefox, so someone may forget about that if they don't read the original instructions.

marcel-bitfly commented on frontend/nuxt.config.ts:
```diff
@@ -81,8 +81,8 @@ export default defineNuxtConfig({
-      cert: 'server.crt',
-      key: 'server.key',
+      cert: 'local.beaconcha.in+3.pem',
+      key: 'local.beaconcha.in+3-key.pem',
```
**question:** is this a typo?

```suggestion
      cert: 'local.beaconcha.in.pem',
      key: 'local.beaconcha.in-key.pem',
```

enzo-bitfly replied:
No, this is the name when you create the certificate for several addresses (`local.beaconcha.in`, `localhost`, `127.0.0.1`,  `::1`), but only `local.beaconcha.in` works, so I'm also changing the creation command in the README to `mkcert local.beaconcha.in`.

marcel-bitfly reviewed (commented):
nice thx for cleaning up and this improvement

enzo-bitfly reviewed (commented):
great review

marcel-bitfly commented on frontend/.gitignore:
```diff
@@ -30,6 +30,10 @@ pnpm-lock.yaml
 
 .npmrc
 
+# SSL certificate files
```
**polish (non-blocking):** 
```suggestion
# SSL certificate files created via `mkcert`
```

this knowledge is also kinda part of the commit message

enzo-bitfly replied:
Done

enzo-bitfly committed:
chore: update config and docs for creating the `server certificate`

- re-enable `certificate` usage in local development with `https` using `mkcert`
- restructure `setup` section in `readme`

See: BEDS-1541
